File: cta-compliant-solution/source/resources/sdk/python/cta_client.py
"""
CTA-5007-B Python SDK
"""

import json
import time
import requests
from typing import Dict, List, Optional

class CTAClient:
    def __init__(self, api_endpoint: str):
        self.api_endpoint = api_endpoint
    
    def generate_token(self, policy: Dict, viewer: Dict = None, media_url: str = None) -> Dict:
        """Generate CTA-5007-B compliant token"""
        payload = {
            'policy': policy,
            'viewer': viewer or {},
            'mediaUrl': media_url
        }
        
        response = requests.post(
            f"{self.api_endpoint}/token",
            json=payload,
            headers={'Content-Type': 'application/json'}
        )
        
        if not response.ok:
            raise Exception(f"HTTP {response.status_code}: {response.text}")
        
        return response.json()
    
    def sign_url(self, media_url: str, policy: Dict, viewer: Dict = None) -> str:
        """Generate signed URL with CTA token"""
        result = self.generate_token(policy, viewer, media_url)
        return result['signedUrl']

# Usage examples
EXAMPLES = {
    'basic': {
        'policy': {'paths': ['/video/'], 'ttl': '2h'},
        'viewer': {'country': 'us'}
    },
    'geo_restricted': {
        'policy': {'paths': ['/premium/'], 'ttl': '24h', 'countries': ['us', 'ca']},
        'viewer': {'country': 'us'}
    }
}

File: cta-compliant-solution/source/resources/sdk/java/CTAClient.java
/**
 * CTA-5007-B Java SDK
 */

import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.net.URI;
import java.util.Map;
import java.util.HashMap;
import com.fasterxml.jackson.databind.ObjectMapper;

public class CTAClient {
    private final String apiEndpoint;
    private final HttpClient httpClient;
    private final ObjectMapper objectMapper;
    
    public CTAClient(String apiEndpoint) {
        this.apiEndpoint = apiEndpoint;
        this.httpClient = HttpClient.newHttpClient();
        this.objectMapper = new ObjectMapper();
    }
    
    public Map<String, Object> generateToken(Map<String, Object> policy, 
                                           Map<String, Object> viewer, 
                                           String mediaUrl) throws Exception {
        Map<String, Object> payload = new HashMap<>();
        payload.put("policy", policy);
        payload.put("viewer", viewer != null ? viewer : new HashMap<>());
        payload.put("mediaUrl", mediaUrl);
        
        String jsonPayload = objectMapper.writeValueAsString(payload);
        
        HttpRequest request = HttpRequest.newBuilder()
            .uri(URI.create(apiEndpoint + "/token"))
            .header("Content-Type", "application/json")
            .POST(HttpRequest.BodyPublishers.ofString(jsonPayload))
            .build();
        
        HttpResponse<String> response = httpClient.send(request, 
            HttpResponse.BodyHandlers.ofString());
        
        if (response.statusCode() != 200) {
            throw new Exception("HTTP " + response.statusCode() + ": " + response.body());
        }
        
        return objectMapper.readValue(response.body(), Map.class);
    }
    
    public String signUrl(String mediaUrl, Map<String, Object> policy, 
                         Map<String, Object> viewer) throws Exception {
        Map<String, Object> result = generateToken(policy, viewer, mediaUrl);
        return (String) result.get("signedUrl");
    }
}
